libhb/work.c

```diff
@@ -1227,6 +1227,33 @@ static int sanitize_audio(hb_job_t *job)
 static int sanitize_qsv( hb_job_t * job )
 {
 #if HB_PROJECT_FEATURE_QSV
+    /* modify filter list while QSV VPP is not implemented
+        Commmand line run usually enables 2 following filters:
+            ID=6 Framerate Shaper (mode=0)
+            ID=12 Crop and Scale (width=1920:height=1080:crop-top=0:crop-bottom=0:crop-left=0:crop-right=0)
+        GUI run usually enables 4 following filters depends on preset:
+            ID=3 Comb Detect (mode=3:spatial-metric=2:motion-thresh=1:spatial-thresh=1:filter-mode=2:block-thresh=40:block-width=16:block-height=16)
+            ID=4 Decomb (mode=39)
+            ID =6 Framerate Shaper (mode=0)
+            ID=12 Crop and Scale (width=1920:height=1080:crop-top=0:crop-bottom=0:crop-left=0:crop-right=0)
+    */
+    if (job->list_filter && hb_list_count(job->list_filter))
+    {
+        // if QSV full is enabled we need to disable filters
+        if(hb_qsv_full_path_is_enabled(job))
+        {
+            for(int filter_id = HB_FILTER_FIRST; filter_id <= HB_FILTER_LAST; filter_id++)
+            {
+                hb_filter_object_t * filter = hb_filter_find(job->list_filter, filter_id);
+                if (filter != NULL)
+                {
+                    hb_list_rem(job->list_filter, filter);
+                    hb_filter_close(&filter);
+                    hb_log("Skip filter with ID=%d", filter_id);
+                }
+            }
+        }
+    }
```

Comment:
I thought the objective was to *disable* qsv full path if any filters are enabled.  That is what I thought the previous patch was trying to do since it only removed filters if they were not really needed.  This new bit of code is unconditionally removing all filters if full path is enabled.

It seems like what should be happening here is something to disable qsv full path if any filters remain after sanitizing the filters.